bakery/poet-cli

```diff
@@ -0,0 +1,290 @@
+    then
+        _say "Skipping '${book_name}' because it does not have an entry in the ABL"
+        return 1
+    fi
+    return 0
+}
+
+do_book() {
+    local book_name=$1
+
+    parse_book_info "${book_name}"
+    return_code=$?
+    [[ ${return_code} == 0 ]] || return ${return_code}
+
+    # At this point the following have been defined:
+    # - book_uuid
+    # - collection_id
+    # - book_style
+    # - book_server
+    generic_args="--data ${temp_dir} ${passthrough_args}"
+    book_col_id=${collection_id}
+    server=${book_server}
+    recipe_name=${book_style}
+    style_file=${book_style}
+
+    for cur_operation in ${all_steps_to_run}
+    do
+        if [[ ${steps_to_run} =~ ${cur_operation} ]]
+        then
+            pointer=dir_for_${cur_operation//-/_} # https://stackoverflow.com/a/55331060 and https://stackoverflow.com/a/5928254
+            test_dir="${temp_dir}/${book_col_id}/${!pointer}"
+            if [[ ${skip_existing} == 0 || ! -d "${test_dir}" ]]
+            then
+                _say "${c_yellow}==>${c_none} ${cur_operation} ${book_name}"
+
+                # Special-case fetch has weird commandline arguments
+                if [[ ${cur_operation} == "${_fetch_arg}" ]]
+                then
+                    try node ./src/cli/execute.js run "${cur_operation}" ${generic_args} "${server}" "${book_col_id}" latest
+                
+                # 'bake' also has weird commandline arguments
+                elif [[ ${cur_operation} == "${_bake_arg}" ]]
+                then
+                    # Verify that the files exist before baking
+                    recipe_file=${CNX_RECIPES_DIR}/recipes/output/${recipe_name}.css
+                    style_file=${CNX_RECIPES_DIR}/styles/output/${recipe_name}-pdf.css
+
+                    [[ -f "${recipe_file}" ]] || die "recipe file not found: ${recipe_file}"
+                    [[ -f "${style_file}" ]] || warn "style file not found. using physics-pdf.css" && sleep 10 && style_file=${CNX_RECIPES_DIR}/styles/output/physics-pdf.css
+                    
+                    try node ./src/cli/execute.js run "${cur_operation}" ${generic_args} "${book_col_id}" "${CNX_RECIPES_DIR}/recipes/output/${recipe_name}.css" "${CNX_RECIPES_DIR}/styles/output/${recipe_name}-pdf.css"
+                
+                # 'link-extras' also has weird commandline arguments
+                elif [[ ${cur_operation} == "${_linkextras_arg}" ]]
+                then
+                    try node ./src/cli/execute.js run "${cur_operation}" ${generic_args} "${book_col_id}" "${server}"
+                else
+                    try node ./src/cli/execute.js run "${cur_operation}" ${generic_args} "${book_col_id}"
+                fi
+
+                # Save the hash of this directory so we can tell if it has changed
+                # & therefore subsequent steps need to re-run
+                pushd "${test_dir}" && find . -type f -exec md5sum {} \; | md5sum > "${test_dir}.dir.md5" && popd
```

Comment:
I'm now getting an error pointing to this line. 
`~/OpenStax/output-producer-service/bakery/temp/col11995/fetched-book ~/OpenStax/output-producer-service/bakery
./poet-cli: line 133: ./temp/col11995/fetched-book.dir.md5: No such file or directory`

```
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module '/Users/bj20/OpenStax/output-producer-service/bakery/temp/col11995/fetched-book/src/cli/execute.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
./poet-cli: ERROR: could not run [node ./src/cli/execute.js run assemble --data ./temp col11995]
```

Reply:
This was after I ran `./poet-cli american-government` 
In `/tmp` I do see `fetched-book`

Reply:
But assembling does work if I run it separately: `./poet-cli -assemble american-government`

Reply:
Another thing to note, the line `find: md5sum: No such file or directory` is printed to my console at least 50+ times after running both `./poet-cli american-government` and `./poet-cli -assemble american-government`